### PR TITLE
Send submissions using ses for file upload

### PR DESCRIPTION
**Pull request by stephencdaly**
### What problem does this pull request solve?

Trello card: https://trello.com/c/JEWV7ew5/

This PR adds a custom delivery method for ActionMailer to send emails using AWS SES. This sends a raw email, which allows us to add attachments. The email uses templates we define in the project to construct the email, which use the `layouts/mailer.html.erb` to apply base formatting to the email.

If a form includes a file upload question, we will send the submission using SES rather than Notify. No live forms currently have the ability to add file upload questions, so for now emails will only be sent using SES when we are doing internal testing.

We are not currently applying GOV.UK styling to the email, this will be done in a subsequent PR which will also look at improving how we format the questions and answers in the email.

We are also not yet attaching uploaded files to the email, which will be done in a subsequent PR. We are however attaching the CSV file for submissions to the email if the form is configured to include this.

This relies on some new environment variables being set, which is done in this PR https://github.com/alphagov/forms-deploy/pull/1231/files

### Testing

Currently, it is necessary to test sending the email using SES by deploying the branch to dev as only the forms-runner role has the permission to send raw emails using SES. We'll want to look at finding an easier way to test soon.

I've tested this on the dev environment and was successfully able to send an email with and without a CSV file attached.

Screenshot 2024-12-17 at 11 01 19

Screenshot 2024-12-17 at 11 11 01

### Things to consider when reviewing

- Ensure that you consider the wider context.
- Does it work when run on your machine?
- Is it clear what the code is doing?
- Do the commit messages explain why the changes were made?
- Are there all the unit tests needed?
- Has all relevant documentation been updated?

### CODE REVIEW

**Comment by chao-xian** [comment]
I was going to ask how to test this locally and noted the PR description that you already tested in Dev.

**Review by chao-xian — DISMISSED** [review]
This is really lovely code @stephencdaly - tysm for your work. Please can you address the small commit history change I requested. Thanks!

File: app/mailers/aws_ses_delivery_method.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-class AwsSESDeliveryMethod
+class AwsSesDeliveryMethod
```
[the remark this answers is not in the document]
Comment (stephencdaly) [reply app/mailers/aws_ses_delivery_method.rb:1]: Good spot, will do this. Thanks!

[the remark this answers is not in the document]
Comment (stephencdaly) [reply app/mailers/aws_ses_delivery_method.rb:1]: Have fixed the history

**Comment by stephencdaly** [comment]
@chao-xian I also just pushed up one more commit with some basic tests for the answer formatting. We'll put some more attention into this for https://trello.com/c/MD1dVOix/2609-update-ses-email-template-to-match-designs-for-file-upload, but I thought it was helpful to make a start.

**Pull request merged**